sklearn/utils/estimator_checks.py

```diff
@@ -1443,41 +1443,90 @@ def check_supervised_y_2d(name, estimator_orig):
     assert_allclose(y_pred.ravel(), y_pred_2d.ravel())
 
 
-@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+@ignore_warnings
+def check_classifiers_predictions(X, y, name, classifier_orig):
+    classes = np.unique(y)
+    classifier = clone(classifier_orig)
+    if name == 'BernoulliNB':
+        X = X > X.mean()
+    set_random_state(classifier)
+    # fit
+    classifier.fit(X, y)
+
+    y_pred = classifier.predict(X)
+
+    if hasattr(classifier, "decision_function"):
+        decision = classifier.decision_function(X)
+        n_samples, n_features = X.shape
+        assert_true(isinstance(decision, np.ndarray))
+        if len(classes) == 2:
+            dec_pred = (decision.ravel() > 0).astype(np.int)
+            dec_exp = classifier.classes_[dec_pred]
+            assert_array_equal(dec_exp, y_pred,
+                               err_msg="decision_function does not match "
+                               "classifier for %r: expected '%s', got '%s'" %
+                               (classifier, ", ".join(map(str, dec_exp)), ", "
```

Comment:
it's clearer with that last "," on the next line, please

Reply:
I tried in the previous commit and I keep getting the error: 'E128 continuation line under-indented for visual indent'

Reply:
I guess this should go in-line with "decision_function doesn't match ", i.e
```python
err_msg="decision_function does not match "
        "classifier for %r ...            "
```
currently "classifier" line starts as:
```python
err_msg="decision_function does not match "
"classifier for %r ...                    "

Reply:
Thank you, @gxyd ! I'll give a try!